### PR TITLE
Part 1 of Database-Driven Product Server assignment. 

### PR DESCRIPTION
**Pull request by aguazul-marco**
Evening Johnny! I figured i'd get some feedback on part 1 of the assignment while I continue to work on part 2. Thank you.

### CODE REVIEW

**Comment by aguazul-marco** [comment]
Copy of the output
`ID: 1
Name: Water - San Pellegrino
Price: 80
ID: 2
Name: Cape Capensis - Fillet
Price: 53
ID: 3
Name: Bread - Bistro Sour
Price: 22
ID: 4
Name: Lamb - Shoulder
Price: 16
ID: 5
Name: Lettuce - Arugula
Price: 11`

File: cmd/seeder/main.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+	Products = products
+
+	for _, p := range Products {
+		fmt.Printf("ID: %v\nName: %v\nPrice: %v\n", p.ID, p.Name, p.Price)
+	}
+
+}
+
+func seed(db *sql.DB) error {
```
Comment (jboursiquot) [inline cmd/seeder/main.go:85]: It would have been better to pass in the list of products as an argument rather than use a global.

**Review by jboursiquot — CHANGES_REQUESTED** [review]
You have things in here that do not belong to this assignment. Clean up and resubmit.

**Comment by aguazul-marco** [comment]
I removed the directories that didn't belong to the assignment and made the updates requested.

File: cmd/seeder/main.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+type Product struct {
+	ID    int     `json:"id"`
+	Name  string  `json:"name"`
+	Price float64 `json:"price"`
+}
+
+var Products []Product
+
+func initProduct() {
```
Comment (jboursiquot) [inline cmd/seeder/main.go:21]: `initProducts` is better name

Comment (jboursiquot) [inline cmd/seeder/main.go:21]: Return the `[]Product` here.
Better yet `func initProducts(path string)([]Product, error)`

File: cmd/seeder/main.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+		log.Fatal(err)
+	}
+
+	if err = json.Unmarshal(data, &Products); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func main() {
+	initProduct()
```
Comment (jboursiquot) [inline cmd/seeder/main.go:33]: Capture the list of products and do something about the possible error returned.

File: cmd/seeder/main.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	if err != nil {
+		log.Fatal(err)
+	}
+	stmt, err := tx.Prepare("INSERT INTO products (name, price) VALUES(?, ?)")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer stmt.Close()
+
+	for _, product := range Products {
```
Comment (jboursiquot) [inline cmd/seeder/main.go:61]: Use lowercase `products`

File: cmd/seeder/main.go [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+			return
+		}
+		products = append(products, p)
+	}
+	err = rows.Err()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	Products = products
```
Comment (jboursiquot) [inline cmd/seeder/main.go:96]: You don't need this

**Review by jboursiquot — CHANGES_REQUESTED** [review]

**Comment by aguazul-marco** [comment]
I updated the initProducts func as well as the variable names we discussed

File: cmd/seeder/main.go [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+	_ "github.com/mattn/go-sqlite3"
+)
+
+type Product struct {
+	ID    int     `json:"id"`
+	Name  string  `json:"name"`
+	Price float64 `json:"price"`
+}
+
+var Products []Product
```
Comment (jboursiquot) [inline cmd/seeder/main.go:19]: You don't need this global.

File: cmd/seeder/main.go [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+	}
+
+	rows, err := db.Query("SELECT id, name, price FROM products LIMIT 5")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer rows.Close()
+
+	var newProducts []Product
```
Comment (jboursiquot) [inline cmd/seeder/main.go:85]: No need to for this. Just print out each product you read from the DB during the for loop over `rows`.

File: cmd/seeder/main.go [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+		newProducts = append(newProducts, p)
+	}
+	err = rows.Err()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, p := range newProducts {
+		fmt.Printf("ID: %v\nName: %v\nPrice: %v\n", p.ID, p.Name, p.Price)
+	}
```
Comment (jboursiquot) [inline cmd/seeder/main.go:103]: No need for this again. You're already looping over the rows. Output during that loop.

**Review by jboursiquot — CHANGES_REQUESTED** [review]

**Comment by aguazul-marco** [comment]
Removed the global []products and for loop at the end.

**Review by jboursiquot — CHANGES_REQUESTED** [review]

**Review by jboursiquot — CHANGES_REQUESTED** [review]
Minor change. Nice work.

**Review by jboursiquot — APPROVED** [review]

**Pull request merged**